addons/project/res_partner.py

```diff
@@ -36,6 +36,7 @@ def _task_count(self, cr, uid, ids, field_name, arg, context=None):
         'task_count': fields.function(_task_count, string='# Tasks', type='integer'),
     }
 
+    @api.cr_uid_id_context
```

Comment:
that was my thinking too. Plus the fact that in python, you *can* pass everything as a keyword argument, so I always try to avoid changing function signatures, even in cases were it's quite improbable for anyone to use the function this way